Celeste.Mod.mm/Mod/UI/TextMenuExt.cs

```diff
@@ -355,5 +355,839 @@ public class IntSlider : TextMenu.Item {
+                item.Container = null;
+                Container.Remove(item.ValueWiggler);
+                Container.Remove(item.SelectWiggler);
+                RecalculateSize();
+                return this;
+            }
+            public void Clear() {
+                Items = new List<TextMenu.Item>();
+            }
+            public int IndexOf(TextMenu.Item item) {
+                return Items.IndexOf(item);
+            }
+            public void FirstSelection() {
+                Selection = -1;
+                MoveSelection(1, false);
+            }
+            public void MoveSelection(int direction, bool wiggle = false) {
+                int selection = Selection;
+                direction = Math.Sign(direction);
+                int num = 0;
+                using (List<TextMenu.Item>.Enumerator enumerator = Items.GetEnumerator()) {
+                    while (enumerator.MoveNext()) {
+                        if (enumerator.Current.Hoverable) {
+                            num++;
+                        }
+                    }
+                }
+                bool flag = num > 2;
+                do {
+                    Selection += direction;
+                    if (flag) {
+                        if (Selection < 0) {
+                            Selection = Items.Count - 1;
+                        } else if (Selection >= Items.Count) {
+                            Selection = 0;
+                        }
+                    } else if (Selection < 0 || Selection > Items.Count - 1) {
+                        Selection = Calc.Clamp(Selection, 0, Items.Count - 1);
+                        break;
+                    }
+                }
+                while (!Current.Hoverable);
+                if (!Current.Hoverable) {
+                    Selection = selection;
+                }
+                if (Selection != selection && Current != null) {
+                    if (selection >= 0 && Items[selection] != null && Items[selection].OnLeave != null) {
+                        Items[selection].OnLeave();
+                    }
+                    Current.OnEnter?.Invoke();
+                    if (wiggle) {
+                        Audio.Play((direction > 0) ? "event:/ui/main/rollover_down" : "event:/ui/main/rollover_up");
+                        Current.SelectWiggler.Start();
+                    }
+                }
+            }
+            public void RecalculateSize() {
+                TitleHeight = ActiveFont.LineHeight;
+                if (Items.Count < 1)
+                    return;
+                LeftColumnWidth = (RightColumnWidth = (MenuHeight = 0f));
+                foreach (TextMenu.Item item in Items) {
+                    if (item.IncludeWidthInMeasurement) {
+                        LeftColumnWidth = Math.Max(LeftColumnWidth, item.LeftWidth());
+                    }
+                }
+                foreach (TextMenu.Item item2 in Items) {
```

Comment:
There's no need for this variable to be named `item2` - `item` in the previous `foreach` isn't conflicting. Same goes for `item3` further down below.